### PR TITLE
victorialogs: add more checks for stats query APIs

### PR DESCRIPTION
**Pull request by Haleygo**
1. Verify if field in [fields pipe](https://docs.victoriametrics.com/victorialogs/logsql/#fields-pipe) exists. If not, it generates a metric with illegal float value "" for prometheus metrics protocol.
2. check if multiple time range filters produce conflicted query time range, for instance:
```
query: _time: 5m | stats count(), 
start:2024-10-08T10:00:00.806Z, 
end: 2024-10-08T12:00:00.806Z, 
time: 2024-10-10T10:02:59.806Z
```
must give no result due to invalid final time range.

### CODE REVIEW

**Review by valyala — APPROVED** [review]

**Comment by valyala** [comment]
@Haleygo , thanks for the pull request!

**Comment by valyala** [comment]
@Haleygo , please see the follow-up commit - 508e498ae3f277f8ac8652fe6c9c161ad3cb0968

File: lib/logstorage/storage_search.go [hunk 1]
```diff
@@ -119,6 +119,9 @@ func (s *Storage) runQuery(ctx context.Context, tenantIDs []TenantID, q *Query,
 	})
 
 	minTimestamp, maxTimestamp := q.GetFilterTimeRange()
+	if minTimestamp > maxTimestamp {
```
[the remark this answers is not in the document]
Comment (valyala) [reply lib/logstorage/storage_search.go:122]: FYI, I removed this check at 508e498ae3f277f8ac8652fe6c9c161ad3cb0968

File: lib/logstorage/parser.go [hunk 2]
```diff
@@ -708,9 +732,20 @@ func ParseQuery(s string) (*Query, error) {
 	return ParseQueryAtTimestamp(s, timestamp)
 }
 
+// ParseStatsQuery parses s with needed stats query checks.
+func ParseStatsQuery(s string) (*Query, error) {
+	timestamp := time.Now().UnixNano()
+	query, err := ParseQueryAtTimestamp(s, timestamp)
```
[the remark this answers is not in the document]
Comment (valyala) [reply lib/logstorage/parser.go:738]: This comment is addressed at 508e498ae3f277f8ac8652fe6c9c161ad3cb0968

File: lib/logstorage/storage_search.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (Haleygo) [reply lib/logstorage/storage_search.go:122]: @valyala Then why not return empty results directly when `minTimestamp > maxTimestamp`, there shouldn't be any data anyway.

**Pull request merged**